pkg/controller/podgc/gc_controller_test.go

```diff
@@ -44,10 +44,12 @@ import (
 func alwaysReady() bool { return true }
 
 func NewFromClient(kubeClient clientset.Interface, terminatedPodThreshold int) (*PodGCController, coreinformers.PodInformer, coreinformers.NodeInformer) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 	informerFactory := informers.NewSharedInformerFactory(kubeClient, controller.NoResyncPeriodFunc())
 	podInformer := informerFactory.Core().V1().Pods()
 	nodeInformer := informerFactory.Core().V1().Nodes()
-	controller := NewPodGC(context.TODO(), kubeClient, podInformer, nodeInformer, terminatedPodThreshold)
+	controller := NewPodGC(ctx, kubeClient, podInformer, nodeInformer, terminatedPodThreshold)
```

Comment:
The context created here would be cancelled on exiting the `NewFromClient` function which is misleading. It does not fail because the context passed to `NewPodGC` is not used in gc_controller: https://github.com/kubernetes/kubernetes/blob/a837be03ffea65115a90b4bc4a8092d9b842971e/pkg/controller/podgc/gc_controller.go#L76. I suggest to either (1) remove the context passage to NewPodGC (preferred, but will require yet a change in cmd/kube-controller-manager) or (2) propagate the context created for each test and pass it to `NewFromClient`, then to `NewPodGC`.